tests/app/dao/test_jobs_dao.py

```diff
@@ -351,17 +351,21 @@ def test_should_get_jobs_seven_days_old_by_scheduled_for_date(sample_service):
 
 @freeze_time("2016-10-31 10:00:00")
 def test_should_get_limited_number_of_jobs(sample_template):
-    flexable_retention_service = create_service(service_name="Another service")
-    insert_service_data_retention(flexable_retention_service.id, sample_template.template_type, 3)
-    flexable_template = create_template(flexable_retention_service, template_type=sample_template.template_type)
+    flexable_retention_service1 = create_service(service_name="Another service 1")
+    insert_service_data_retention(flexable_retention_service1.id, sample_template.template_type, 3)
+    flexable_template1 = create_template(flexable_retention_service1, template_type=sample_template.template_type)
+
+    flexable_retention_service2 = create_service(service_name="Another service 2")
+    insert_service_data_retention(flexable_retention_service2.id, sample_template.template_type, 3)
+    flexable_template2 = create_template(flexable_retention_service2, template_type=sample_template.template_type)
 
     eight_days_ago = datetime.utcnow() - timedelta(days=8)
     four_days_ago = datetime.utcnow() - timedelta(days=4)
 
-    create_job(flexable_template, created_at=four_days_ago)
-    create_job(flexable_template, created_at=four_days_ago)
-    create_job(sample_template, created_at=eight_days_ago)
-    create_job(sample_template, created_at=eight_days_ago)
+    for _ in range(4):
+        create_job(flexable_template1, created_at=four_days_ago)
+        create_job(flexable_template2, created_at=four_days_ago)
+        create_job(sample_template, created_at=eight_days_ago)
```

Comment:
Tweaked this test so that in the old PR it would add limit=3 jobs for both service1 and service2, then in the 7 day query it would try to set the query limit to 3 - 6 = -3 and throw the error we saw in staging.

The fixed code will add 3 jobs for service1, then the queries for service2 and the 7-day query will have limit zero.